federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -307,71 +652,92 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 			}
 		} else {
 			clusterIngress := clusterIngressObj.(*extensions_v1beta1.Ingress)
-			glog.V(4).Infof("Found existing Ingress %s in cluster %s - checking if update is required", ingress, cluster.Name)
-			clusterIPName, clusterIPExists := clusterIngress.ObjectMeta.Annotations[staticIPAnnotationKey]
-			if !baseIPExists && clusterIPExists {
-				// Add annotation to federated ingress via API.
-				original, err := ic.federatedApiClient.Extensions().Ingresses(baseIngress.Namespace).Get(baseIngress.Name)
-				if err == nil {
-					original.ObjectMeta.Annotations[staticIPAnnotationKey] = clusterIPName
-					if _, err = ic.federatedApiClient.Extensions().Ingresses(baseIngress.Namespace).Update(original); err != nil {
-						glog.Errorf("Failed to add static IP annotation to federated ingress %q: %v", ingress, err)
+			glog.V(4).Infof("Found existing Ingress %s in cluster %s - checking if update is required (in either direction)", ingress, cluster.Name)
+			clusterIPName, clusterIPNameExists := clusterIngress.ObjectMeta.Annotations[staticIPNameKeyReadonly]
+			baseLBStatusExists := len(baseIngress.Status.LoadBalancer.Ingress) > 0
+			clusterLBStatusExists := len(clusterIngress.Status.LoadBalancer.Ingress) > 0
+			logStr := fmt.Sprintf("Cluster ingress %q has annotation %q=%q, loadbalancer status exists? [%v], federated ingress has annotation %q=%q, loadbalancer status exists? [%v].  %%s annotation and/or loadbalancer status from cluster ingress to federated ingress.", ingress, staticIPNameKeyReadonly, clusterIPName, clusterLBStatusExists, staticIPNameKeyWritable, baseIPName, baseLBStatusExists)
+			if (!baseIPAnnotationExists && clusterIPNameExists) || (!baseLBStatusExists && clusterLBStatusExists) { // copy the IP name from the readonly annotation on the cluster ingress, to the writable annotation on the federated ingress
+				glog.V(4).Infof(logStr, "Transferring")
+				if !baseIPAnnotationExists && clusterIPNameExists {
+					baseIngress.ObjectMeta.Annotations[staticIPNameKeyWritable] = clusterIPName
+				}
+				if !baseLBStatusExists && clusterLBStatusExists {
+					lbstatusObj, lbErr := conversion.NewCloner().DeepCopy(&clusterIngress.Status.LoadBalancer)
+					lbstatus, ok := lbstatusObj.(*v1.LoadBalancerStatus)
+					if lbErr != nil || !ok {
+						glog.Errorf("Internal error: Failed to clone LoadBalancerStatus of %q in cluster %q while attempting to update master loadbalancer ingress status, will try again later. error: %v, Object to be cloned: %v", ingress, cluster.Name, lbErr, lbstatusObj)
+						ic.deliverIngress(ingress, ic.ingressReviewDelay, true)
+						return
 					}
+					baseIngress.Status.LoadBalancer = *lbstatus
+				}
+				glog.V(4).Infof("Attempting to update base federated ingress: %v", baseIngress)
+				if _, err = ic.federatedApiClient.Extensions().Ingresses(baseIngress.Namespace).Update(baseIngress); err != nil {
```

Comment:
Ideally this needs to be done in parallel via a worker queue.  Will do in a followup PR.